### PR TITLE
Many clients revisited

### PR DESCRIPTION
**Pull request by Munksgaard**
Uses the fancy `<Server as HasDual>::Dual` trick! Love it!

### CODE REVIEW

**Comment by laumann** [comment]
Magic, indeed! :godmode:

File: examples/many-clients.rs [hunk 1]
```diff
@@ -7,9 +7,9 @@ use std::thread::spawn;
 use rand::random;
 
 type Server = Recv<u8, Choose<Send<u8, Eps>, Eps>>;
-type Client = Send<u8, Offer<Recv<u8, Eps>, Eps>>;
+type Client = <Server as HasDual>::Dual;
```
Comment (laumann) [inline examples/many-clients.rs]: :heart:

**Comment by laumann** [comment]
Is this enough argument to remove accept/request? And if so, should we do it?

**Comment by laumann** [comment]
r+ by the way

**Comment by Munksgaard** [comment]
For reference, we are talking about 570871f specifically.

I'm still not quite sure. I am now convinced that we don't _need_ accept/request any more. However, they might be nice utility functions to have? Using `accept` does decrease the line count slightly, but I guess users who find themselves using it a lot could just roll their own? It makes no difference at all for the server, it seems.

File: examples/many-clients.rs [hunk 2]
```diff
@@ -20,18 +20,18 @@ fn handler(c: Chan<(), Server>) {
 fn server(rx: Receiver<Chan<(), Server>>) {
     let mut count = 0;
     loop {
-        match borrow_request(&rx) {
-            Some(c) => {
-                spawn(move || handler(c));
+        match rx.recv() {
+            Ok(c) => {
+                spawn(move || server_handler(c));
```
Comment (Manishearth) [inline examples/many-clients.rs]: Is the move annotation still necessary?

Comment (Munksgaard) [inline examples/many-clients.rs]: It seems like I can remove this particular annotation, but not the one in the `main` function. Any idea why?

The error I'm getting is this:

```
examples/many-clients.rs:56:9: 56:14 error: the trait `core::marker::Sync` is not implemented for the type `core::cell::UnsafeCell<std::sync::mpsc::Flavor<session_types::Chan<(), session_types::Recv<u8, session_types::Choose<session_types::Send<u8, session_types::Eps>, session_types::Eps>>>>>` [E0277]
examples/many-clients.rs:56         spawn(|| {
                                    ^~~~~
examples/many-clients.rs:56:9: 56:14 note: `core::cell::UnsafeCell<std::sync::mpsc::Flavor<session_types::Chan<(), session_types::Recv<u8, session_types::Choose<session_types::Send<u8, session_types::Eps>, session_types::Eps>>>>>` cannot be shared between threads safely
examples/many-clients.rs:56         spawn(|| {
                                    ^~~~~
error: aborting due to previous error
```

**Pull request merged**